src/Libraries/CoreNodeModels/Input/BaseTypes.cs

```diff
@@ -131,11 +131,11 @@ public override NodeInputData InputData
            get {
                 return new NodeInputData()
                 {
-                    Id = this.GUID.ToString("N"),
+                    Id = this.GUID,
                     Name = this.Name,
                     Type = NodeInputTypes.numberInput,
                     Description = this.Description,
-                    Value = Value.ToString(),
+                    Value = System.Convert.ToDouble(Value),
```

Comment:
we do this to let the json serialization take care of the conversion based on type.